src/app.ts

```diff
@@ -66,6 +76,20 @@ app.get("/price", cache("10 minutes"), async (req, res) => {
   res.send(price);
 });
 
+app.get("/circulating-supply", async (req, res) => {
```

Comment:
I suggest we delay the app.listen until we guarantee we have fetched the circulating supply so we don't have to deal with two cases. So if file exists we can send a response otherwise we return a internal server error 500. (Same applies to carousel-data)

Secondly I think the response expected by CMC for this endpoint is a plain text so we can do this instead:

```ts
res.setHeader("Content-Type", "text/plain");
const { circulatingSupply } = JSON.parse(circulatingSupplyData.toString())
res.send(`${circulatingSupply}`).end();
```

Reply:
note the format is because `circulatingSupply` is a `number`, and would trigger an express error as it would interpret it as a HTTP status response  code.